drafts/20003.adoc

```diff
@@ -0,0 +1,457 @@
+* <<ejb, Persistent EJB Timers coordination and failover across members (ejbPersistentTimer-3.2)>>
+
+These early implementations are not available in 20.0.0.3 but you can try them out in our daily Docker image by running `docker pull openliberty/daily`. Let us know what you think!
+
+[#mp3]
+== MicroProfile 3.3 support
+[#mpr]
+=== MicroProfile Rest Client (mpRestClient-1.4)
+
+MicroProfile Rest Client 1.4 adds injection into ClientHeadersFactory instances. When executing a Rest Client inside a JAX-RS request, it can be useful to pull data from the JAX-RS request's context or from CDI to use to determine which HTTP headers to send on the outgoing request. With MP Rest Client 1.4, this is now possible.
+
+To enable MP Rest Client 1.4, add this feature to your server.xml:
+`<feature>mpRestClient-1.4</feature>`
+
+CDI and/or JAX-RS injection into your ClientHeadersFactory will enable you to do things like:
+
+[source, java]
+----
+@ApplicationScoped
+public class MyCustomClientHeadersFactory implements ClientHeadersFactory {
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Inject
+    private Foo foo;
+
+    @Override
+    public MultivaluedMap<String, String> update(MultivaluedMap<String, String> incomingHeaders,
+                                                 MultivaluedMap<String, String> clientOutgoingHeaders) {
+        MultivaluedMap<String, String> myHeaders = new MultivaluedHashMap<>();
+        myHeaders.putSingle("X-HEADER_FROM_CUSTOM_CLIENTHEADERSFACTORY", "456");
+
+        URI uri = uriInfo.getAbsolutePath();
+        myHeaders.putSingle("X-INJECTED_URI_INFO", uri == null ? "null" : uri.toString());
+
+        myHeaders.putSingle("X-INJECTED_FOO", foo.getWord());
+
+        return myHeaders;
+    }
+}
+----
+link:https://github.com/eclipse/microprofile-rest-client[Learn more about MP Rest Client]
+
+[#mra]
+=== Monitor microservice applications easily wth metrics (mpMetrics-2.3)
+
+MicroProfile Metrics 2.3 introduces a new metric type called a Simple Timer (annotated with `@SimplyTimed`) and runtime provided metrics that track REST resource method usage and is backed by the new Simple Timer metric.
+
+The new Simple Timer metric is a light-weight alternative to the existing Timer metric. It only tracks the total timing duration and counts the amount of times it was invoked. The Timer metric on the other hand is a performance heavy metric that continually calculates duration statistics and throughput statistics resulting in 14 values.
+
+The new REST stat metrics are gathered from REST resource method usage (i.e `GET`, `POST`, `PUT`, `DELETE`, `OPTIONS`, `PATCH`, `HEAD`). Total time duration and total count of invocation is tracked ( by use of the Simple Timer metric). This functionality is properly enabled when used in combination with the `jaxrsMonitor-1.0` feature. ALL REST stat metrics will use the REST.request metric name and will be tagged/labeled with their fully qualified class name and method signature.
```

Comment:
"This functionality is properly enabled when used in combination with the `jaxrsMonitor-1.0` feature."

This statement is accurate for the beta, but will need to change for GA.  For GA (currently targeting 20.0.0.4), it should read:
"This functionality is properly enabled when both the `mpMetrics-2.3` and `jaxrs-2.1` features are enabled."